Automattic/sensei
# Add base button block

Provide a button block base that can be used for the upcoming button blocks.

### Changes proposed in this Pull Request

* Move block registration and entry files up a directory, as `blocks/index.js` and  `blocks/single-course.scss` for now.
* Add a Button builder function that contains an extendable base block definition
* Add a simple Take course button to use this
* Add the block radius code from Gutenberg core's button component
* Add support for for the new built-in color support key
  * This adds color settings, including gradients, and style rendering
  * We then remove the style rendering, because it applies the style to the block wrapper element :)
  * On older Wordpress versions, fall back to other color settings helpers we already use for the outline blocks. (No gradients)
* Add alignment support
  * Full width default is only for the Take course button, but it's just CSS
* Add fill and outline block styles (uses built-in styling)

### Testing instructions

* Create a course and add a Take Course button
* Customize it, save & publish/preview course
* Should look the same on the frontend
* Button should work and retain all settings after reloading the editor
* Button should mostly look and work the same as the core button across various themes

### TODO


### Screenshot / Video

<img width="629" alt="image" src="https://user-images.githubusercontent.com/176949/95376502-78a9f480-08e1-11eb-8a6e-ee4761d2f2c0.png" align="center">

<img width="820" alt="image" src="https://user-images.githubusercontent.com/176949/95376763-ddfde580-08e1-11eb-9bea-75784b515cf8.png" align="center">
